source/slang/slang-parser.cpp

```diff
@@ -5579,20 +5587,75 @@ namespace Slang
         return varDeclrStatement;
     }
 
+    static void desugarIfLetStatement(IfLetStmt* ifletStatement, Parser* parser)
```

Comment:
Here we should just create and return a new SeqStmt consisting of a DeclStmt and a IfStmt.

After lowering, there should not be any IfLet left.

Reply:
Done